### PR TITLE
Add OIDC Client SPI

### LINKED ISSUE #42878: Add OIDC Client SPI
### Description

Some Quarkus and custom extensions may need to acquire OIDC tokens to complete some of their actions. 
OIDC tokens are acquired with OIDC/OAuth2 token grants, some of these grants are rather complex to implement, for example, a JWT bearer token grant supported by Google API - it requires creating a signed authentication JWT with a private key and using it to acquire a token. Besides, a process of refreshing tokens can also be quite involved.

OIDC client has been designed to do it all for the users, but extensions which may want to depend on OIDC client to do its work may not want to add a complete OIDC client dependency.

Therefore creating a single interface SPI will let such extensions depend on OIDC client indirectly with users adding `quarkus-oidc-client` themselves as necessary.

The reason I started thinking about is that this is what Mailer will have to do, to acquire and refresh tokens without the user been involved to support system emails.

However, I believe we have at least one pending rather old PR related to one of the other extensions which requires OIDC client to acquire tokens, and IMHO, several other extensions can benefit, even we decide not to bring this SPI into the Mailer

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by sberyozkin**
Fixes #42878

### CODE REVIEW

**Comment by sberyozkin** [comment]
I know we are supposed to try to close smaller issues, but I'd like to make sure we have a good support in LTS fore the forthcoming Mailer XOAuth2 work...

File: extensions/oidc-client/spi/src/main/java/io/quarkus/oidc/client/spi/TokenProvider.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package io.quarkus.oidc.client.spi;
+
+import io.smallrye.mutiny.Uni;
+
+public interface TokenProvider {
```
Comment (gastaldi) [inline extensions/oidc-client/spi/src/main/java/io/quarkus/oidc/client/spi/TokenProvider.java:5]: Can you add a Javadoc to this interface explaining its purpose?

**Review by gastaldi — APPROVED** [review]

File: extensions/oidc-client/spi/src/main/java/io/quarkus/oidc/client/spi/TokenProvider.java (hunk 1 above)
Comment (sberyozkin) [reply extensions/oidc-client/spi/src/main/java/io/quarkus/oidc/client/spi/TokenProvider.java:15]: Hi @gastaldi Added a Javadoc, thanks

**Review by cescoffier — APPROVED** [review]

**Comment by sberyozkin** [comment]
Let me merge as I've got 2 approvals, Michal, @michalvavrik, ping me please if you'd like to add anything

**Review by michalvavrik — COMMENTED** [review]
I think this makes sense. LGTM

**Pull request merged**